keps/sig-apps/3973-consider-terminating-pods-deployment/README.md

```diff
@@ -0,0 +1,752 @@
+The number of calls may be higher in some scenarios if a large `spec.strategy.rollingUpdate.maxSurge`
+value is specified. But the maximum number of calls per deployment should be similar to when
+`spec.strategy.rollingUpdate.maxSurge` value is set to 1 when the feature is disabled.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes.
+
+- API: ReplicaSet
+  Estimated increase in size:
+    - New field in ReplicaSet status about 4 bytes.
+- API: Deployment
+  Estimated increase in size:
+    - New field in Deployment spec about 11 bytes.
+    - New field in Deployment status about 4 bytes.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- 2023-05-01: First version of the KEP opened (https://github.com/kubernetes/enhancements/pull/3974).
+- 2023-12-12: Second version of the KEP opened (https://github.com/kubernetes/enhancements/pull/4357).
+
+## Drawbacks
+
+Deployment rollouts may be slower when using this feature.
+
+## Alternatives
```

Comment:
It may be obvious to you but why are we not considering adding similar patterns for the other app controllers (StatefulSet, DaemonSet, ReplicationController)

Reply:
Good idea, I have added this to the `Notes/Constraints/Caveats (Optional)` section. I am keeping the `Alternatives` section, only for the alternatives to this feature.